File: ios/common/HyperTrackError.swift
enum HyperTrackError: String {
    case gpsSignalLost
    case locationMocked
    case locationPermissionsDenied
    case locationPermissionsInsufficientForBackground
    case locationPermissionsNotDetermined
    case locationPermissionsReducedAccuracy
    case locationPermissionsProvisional
    case locationPermissionsRestricted
    case locationServicesDisabled
    case locationServicesUnavailable
    case motionActivityPermissionsNotDetermined
    case motionActivityPermissionsDenied
    case motionActivityServicesDisabled
    case motionActivityPermissionsRestricted
    case networkConnectionUnavailable
    case invalidPublishableKey
    case blockedFromRunning
}

File: android/src/main/java/com/reactnativehypertracksdk/common/HyperTrackError.kt
package com.reactnativehypertracksdk.common

/**
 * Check HyperTrack SDK docs for HyperTrackError
 * Enum naming convention is ignored to make datatype sync across platforms easier
 */
@Suppress("EnumEntryName")
internal enum class HyperTrackError {
    gpsSignalLost,
    locationMocked,
    locationPermissionsDenied,
    locationPermissionsInsufficientForBackground,
    locationPermissionsNotDetermined,
    locationPermissionsReducedAccuracy,
    locationPermissionsProvisional,
    locationPermissionsRestricted,
    locationServicesDisabled,
    locationServicesUnavailable,
    motionActivityPermissionsNotDetermined,
    motionActivityPermissionsDenied,
    motionActivityServicesDisabled,
    motionActivityServicesUnavailable,
    motionActivityPermissionsRestricted,
    networkConnectionUnavailable,
    invalidPublishableKey,
    blockedFromRunning
}
